### ISSUE TITLE
VS Code + WSL: Command 'Preview AsyncAPI' resulted in an error

### ISSUE THREAD

**Issue by davidlday** [asyncapi/vs-asyncapi-preview#28]
#### Describe the bug

When using from within WSL on Windows 10, I get the following error:

![image](https://user-images.githubusercontent.com/1132144/129227296-30176a2c-ebd1-4b25-b57d-43d65189b212.png)

#### How to Reproduce

Steps to reproduce the issue:

- in VS Code, open a project from inside WSL that contains an AsyncAPI spec
- in VS Code, open the AsyncAPI spec document
- in VS Code, open the command palette and select "Preview AsyncAPI"

#### Expected behavior

I expect the preview to show a preview of the AsyncAPI spec document

**Comment by Max-Liuhu** [comment]
try to install node, then use  AsyncAPI Preview

**Comment by davidlday** [comment]
node is part of the VSCode environment:
```
Version: 1.59.1
Commit: 3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
Date: 2021-08-19T11:56:46.957Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by MiraIsa** [comment]
having same issue.

```
Version: 1.60.2 (user setup)
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T12:00:31.514Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by davidlday** [comment]
> Please follow instructions in #3
> 
> Tl;dr You should have installed `v12.16+ < Node.js` and `v6.13.7+ < npm` on your computer.

@voborgus - the implication is that the extension will use a version of Node other than the version provided by VS Code, but I don't think that's the case. In fact, I use `nodenv` to manage multiple installs, have tried setting both my project and the extension to use `v12.16.3`, and have had the same results. #3 makes no mention of the versions of VS Code at play, like the three comments above do.

What am I missing? How do I tell an extension to use a different version of Node than the one embedded in VS Code?

Also, `@asyncapi/generator` v1.18.18 is out (see: https://github.com/asyncapi/vs-asyncapi-preview/pull/61). I'm wondering if the fix ultimately is to have this upgraded to be compatible with the version of Node included in more recent versions of VS Code.

**Comment by agrz** [comment]
I have the same issue.
Checking the logs it looks like it is because of some incompatibility with newer npm versions:
`[2021-11-19 18:11:59.659] [exthost] [error] Activating extension ivangsa.asyncapi-preview failed due to an error:
[2021-11-19 18:11:59.659] [exthost] [error] Error: The programmatic API was removed in npm v8.0.0
	at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\index.js:4:9)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\node_modules\global-npm\index.js:29:15
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\node_modules\global-npm\index.js:41:3)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\node_modules\npmi\npmi.js:1:76)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\node_modules\@asyncapi\generator\lib\generator.js:14:14)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\out\PreviewServer.js:20:27)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\out\extension.js:16:25)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at v (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:698)
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ivangsa.asyncapi-preview-0.1.1\out\index.js:4:19)
	at Module.s._compile (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:4:1315)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
	at Module.load (internal/modules/cjs/loader.js:982:32)
	at Module._load (internal/modules/cjs/loader.js:823:14)
	at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
	at Function.i._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:28495)
	at Function.b._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:25083)
	at Function.d._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:103:60224)
	at Module.require (internal/modules/cjs/loader.js:1006:19)
	at require (internal/modules/cjs/helpers.js:88:18)
	at Function.r [as __$__nodeRequire] (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:5:101)
	at w._loadCommonJSModule (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:26492)
	at w._doActivateExtension (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:99:12722)
	at w._activateExtension (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:99:11695)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async Promise.all (index 0)`

I have node 17.0.1 and npm 8.1.0 installed. It looks like the extension does not use the VS code supplied node version (which is 14.16 and should have npm 6.14.1).
I tried to manually downgrade the npm version but still got the same error. So I am not 100% sure how the extension finds node and npm. Updating the asyncapi/generator as @davidlday  suggested may solve the problem as it looks from the stacktrace the problem is caused by the generator dependency.

**Comment by ksilverwall** [comment]
I have same problem on Intel Mac

```
Command 'Preview AsyncAPI' resulted in an error (command 'asyncapi.preview' not found)
```

<img width="262" alt="スクリーンショット 2022-01-31 17 05 04" src="https://user-images.githubusercontent.com/10096645/151758798-caa52623-43d2-4f4a-8fb9-4f6188658cd0.png">

**Comment by RFRIEDM-Trimble** [comment]
Facing same issue, Native Ubuntu 20.04.

**Comment by ivangsa** [comment]
Hi @davidlday 

AsyncAPI vscode extension has been reworked using the latest [@asyncapi/asyncapi-react](https://github.com/asyncapi/asyncapi-react/tree/next) which is also de component used on  AsyncAPI Studio

It has also been relocated but it will autocatically upgrade itself if you already have it installed 
https://marketplace.visualstudio.com/items?itemName=asyncapi.asyncapi-preview 

I'm closing this issue as it should be fixed now.. but please recreate it if this issue still persist.

Thanks for your patience
